src/Mono.Android/Android.App/SyncContext.cs

```diff
@@ -14,7 +15,15 @@ public override SynchronizationContext CreateCopy ()
 
 		public override void Post (SendOrPostCallback d, object state)
 		{
-			using (Handler h = new Handler(Application.Context.MainLooper)) {
+			var looper = Application.Context?.MainLooper;
+			if (looper == null) {
+				var message = "No Looper available to this SynchronizationContext, callback will not be executed.";
```

Comment:
It will almost certainly be some bizarro autogenerated name, but what other choice do we have? "Some method is trying to use the SynchronzationContext, but we won't provide any clue as to what that is"? While accurate, that's...not entirely helpful. `d.Method.DeclaringType.FullName`, will *at least* provide *some* context, and frequently the method name has the "declaring" method as an embedded substring of `d.Method.Name`, so it's at least *something* to guide the developer to the source of the `SynchronizationContext.Post()` invocation...